src/main/java/net/snowflake/client/jdbc/SnowflakeFileTransferAgent.java

```
@@ -1780,9 +1783,20 @@ private void downloadFiles() throws SnowflakeSQLException {
       try {
         // wait for all threads to complete without timeout
         threadExecutor.awaitTermination(Long.MAX_VALUE, TimeUnit.DAYS);
+        for (Future<Void> downloadFileFuture : downloadFileFutures) {
+          if (downloadFileFuture.isDone()) {
+            downloadFileFuture.get();
+          }
+        }
       } catch (InterruptedException ex) {
         throw new SnowflakeSQLLoggedException(
             queryID, session, ErrorCode.INTERRUPTED.getMessageCode(), SqlState.QUERY_CANCELED);
+      } catch (ExecutionException ex) {
```

can we have the tests for get and put where we can see suche exception?

---

We already had a test checking if injected exception was thrown (https://github.com/snowflakedb/snowflake-jdbc/blob/master/src/test/java/net/snowflake/client/jdbc/FileUploaderLatestIT.java#L435) but it was also failing silently due to asserts being only in catch block which never executed. I'll fix such tests. Regarding real life scenario it's hard to say. One example is this invalid path when using `file:///` prefix but I'm about to patch it. Other possible exception is file being removed between initial check if it exists called from the main thread and the execution of callable. But I guess fixing the test with injected one will be sufficient